### PR TITLE
Refactor hyperdrive-wasm build script.

### PR DESCRIPTION
**Pull request by sentilesdal**
- copy build script
- add wasm-pkg-build
- use async/await
- elevate error handling
- don't worry about backups for now
- add cargo.toml
- refactor build script
- update package files

### CODE REVIEW

File: crates/hyperdrive-wasm/scripts/build2.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    JSON.stringify(packageJson, null, 2),
+  );
+
+  // Release the lockfile.
+  release();
+
+  console.log("Done!");
+
+  // Ensure the lockfile is released.
+  lockfile.unlock(__filename);
```
Comment (ryangoree) [inline crates/hyperdrive-wasm/scripts/build2.ts]: We can remove this like now.

File: crates/hyperdrive-wasm/scripts/build2.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  console.log("Removing temporary build files...");
+  rmSync(DEFAULT_TMP_DIR, { recursive: true });
+
+  console.log("Creating package.json...");
+
+  // Load the Cargo.toml file
+  const cargoToml = tomlJson({
+    fileUrl: "Cargo.toml",
+  }) as CargoToml;
+
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply crates/hyperdrive-wasm/scripts/build2.ts:151]: oh yeah great idea

File: crates/hyperdrive-wasm/scripts/build2.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  // Now we'll create the package.json.
+  const packageJson = {
+    name: `@delvtech/${cargoToml.package.name}`,
+    collaborators: cargoToml.package.authors.filter(Boolean),
+    version: cargoToml.package.version,
+    license: cargoToml.package.license,
+    files: fileNames.map(({ to }) => to),
+    module: "hyperdrive_wasm.js",
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply crates/hyperdrive-wasm/scripts/build2.ts:159]: thx forgot

File: crates/hyperdrive-wasm/scripts/build2.ts [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+      to: `${filePrefix}.d.ts`,
+    }, // types
+    {
+      from: `${filePrefix}_bg.wasm`,
+      to: `${filePrefix}_bg.wasm`,
+    }, // wasm for cjs
+  ];
+  fileNames.forEach(({ from, to }) => {
+    renameSync(resolve(TMP_DIR, from), resolve(outDir, to));
+  });
```
Comment (ryangoree) [inline crates/hyperdrive-wasm/scripts/build2.ts:127]: 🔥

File: packages/hyperdrive-js/src/hyperwasm.ts [hunk 5]
```diff
@@ -1,5 +1,4 @@
+// eslint-disable-next-line import/no-unresolved
 import * as hyperwasm from "@delvtech/hyperdrive-wasm";
```
Comment (ryangoree) [inline packages/hyperdrive-js/src/hyperwasm.ts:2]: The types don't resolve? We should definitely fix this before merging.

**Review by ryangoree — APPROVED** [review]

**Pull request merged**